test/e2e/framework/rs_util.go

```diff
@@ -138,3 +140,99 @@ func RunReplicaSet(config testutils.ReplicaSetConfig) error {
 	config.ContainerDumpFunc = LogFailedContainers
 	return testutils.RunReplicaSet(config)
 }
+
+func NewReplicaSet(name, namespace string, replicas int32, podLabels map[string]string, imageName, image string) *extensions.ReplicaSet {
+	return &extensions.ReplicaSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ReplicaSet",
+			APIVersion: "extensions/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: extensions.ReplicaSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: podLabels,
+			},
+			Replicas: &replicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: podLabels,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  imageName,
+							Image: image,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func LogReplicaSet(rs *extensions.ReplicaSet) {
+	if rs != nil {
+		Logf(spew.Sprintf("Replicaset %q:\n%+v", rs.Name, *rs))
+	} else {
+		Logf("Replicaset is nil.")
+	}
+}
+
+// WaitForReplicaSetImage waits for the RS's container image to match the given image.
+func WaitForReplicaSetImage(c clientset.Interface, ns, replicaSetName string, image string, pollInterval, pollTimeout time.Duration) error {
```

Comment:
This doesn't make sense for ReplicaSet, because unlike Deployment, it doesn't do rolling updates.

Reply:
Noted - I was not aware that this function was relevant to rolling updates. In fact, any change to pod template should be good (not necessarily the image).